src/Identity/IdentityServer/ClientStore.cs

```diff
@@ -100,7 +100,7 @@ private async Task<Client> CreateApiKeyClientAsync(string clientId)
         {
             ClientId = clientId,
             RequireClientSecret = true,
-            ClientSecrets = { new Secret(apiKey.ClientSecret.Sha256()) },
+            ClientSecrets = { new Secret(apiKey.HashedClientSecret) },
```

Comment:
You will want to check if the value is null, and fall back to the old value.

Reply:
Maybe I'm missing something, but from what I understood from the EDD process, this application code needs to work both for the transition migration and future migration. 

That would mean we wouldn't have access to the `ClientSecret` column?

Reply:
After re-reading things in the EDD docs again, I think I misunderstood the order of how these would be applied.

The order is?:
- schema changes (adding the column in this case)
- server application code release
- data migration ran on database
- future migration ran on database

Assuming that is the case, I've changed things to work with this order of operations. 

Mainly needed to add the `ClientSecret` property to `ApiKeyDetails` so we can have the client secret on this read operation in the `ClientStore`. 

I'm guessing after release, the idea is to make another PR to take this property out of `ApiKeyDetails`?

Reply:
Yes, and to move the future migration to current.